### PR TITLE
Migrating to PackageReference

### PR DESCRIPTION
**Pull request by Neme12**
The first commit migrates existing packages from packages.config to PackageReference. I did this manually as opposed to using the automatic migration:
image
and only added top-level packages, leaving out all transitive dependencies, even those that were included explicitly with a different version. This means that some transitive dependencies are now using a different minor version, but I hope that's fine. Otherwise we can use the automatic migration which would preserve versions of all packages while still trimming down explicit package references as much as possible.

The second commit adds back Microsoft.VisualStudio.SDK.EmbedInteropTypes, which should be necessary until https://github.com/NuGet/Home/issues/2365 is resolved. But I noticed this packages was removed in ea33ed1f7dc8afc7cc45f7dcdb10d257232cbbff. Is there a reason for that?

### CODE REVIEW

File: TemporaryProjects/TemporaryProjects.csproj [hunk 1]
```diff
@@ -35,6 +32,7 @@
     <StartAction>Program</StartAction>
     <StartProgram Condition="'$(DevEnvDir)' != ''">$(DevEnvDir)devenv.exe</StartProgram>
     <StartArguments>/rootsuffix Exp</StartArguments>
+    <RestoreProjectStyle>PackageReference</RestoreProjectStyle>
```
Comment (Neme12) [inline TemporaryProjects/TemporaryProjects.csproj]: This is not strictly necessary when this project already contains PackageReference - VS will always keep using the current format (packages.config vs PackageReference). This only ensures that if all packages were removed, VS would choose PackageReference as the default.

**Comment by Neme12** [comment]
Visual Studio seems to be adding some extra packages when using New Item -> Extensibility. But then I checked master and it's actually doing the same thing already after ea33ed1f7dc8afc7cc45f7dcdb10d257232cbbff and trying to add back some 15.0 packages. Is there a workaround for that?

File: TemporaryProjects/TemporaryProjects.csproj (hunk 1 above)
Comment (jcansdale) [reply TemporaryProjects/TemporaryProjects.csproj]: Thanks for the tip.

**Comment by jcansdale** [comment]
> ...and only added top-level packages, leaving out all transitive dependencies

This is great! I hate having to wrestle with the old NuGet interface and transitive dependencies.

> But I noticed this packages was removed in ea33ed1. Is there a reason for that?

Probably me getting frustrated with NuGet and starting to hit buttons randomly. 😉 

There is actually a situation where I prefer interop types to *not* be embed.

There's a feature in TestDriven.Net (`Test With > In-Proc`) that lets me execute the following method and have it automatically pass in the service object:
```cs
        [STAThread]
        void ShowActiveView(IVsTextManager textManager)
        {
            ErrorHandler.ThrowOnFailure(textManager.GetActiveView(1, null, out IVsTextView activeView));
            activeView.GetCaretPos(out int line, out int col);
            Console.WriteLine($"Caret is on line {line}");
        }
```

In order for this to work it needs to guess the service type from the interface type. In this case `IVsTextManager` -> `SVsTextManager`. The problem comes when it needs to find the assembly for the service type. If `IVsTextManager` isn't embedded, it can look at the assembly it comes from and guess where to find the service type. The problem is when IVsTextManager` is embedded, it no longer has this hint about where to find the service type. 😕 

Luckily this generally isn't a problem and I can disable embedding when I run into it. I didn't know about the whole discussion around https://github.com/NuGet/Home/issues/2365. I'll dig into this some more.

**Comment by jcansdale** [comment]
@Neme12, I hope you don't mind, but I've taken the liberty of pushing to your PR branch. I've updated the project to use `EnvDTE` from NuGet and removed a few unused references.

**Comment by Neme12** [comment]
@jcansdale Feel free to modify this :smile: I don't have such a level of knowledge in COM interop or VS extension development to understand all of this in detail (I just started experimenting).

**Comment by jcansdale** [comment]
Thanks for submitting this PR and pointing to the `Microsoft.VisualStudio.SDK.EmbedInteropTypes` package and related issue.

This has flushed out and issue with TestDriven.Net that I'd really like to get fixed. I've created an issue for it here: https://github.com/jcansdale/TestDriven.Net-Issues/issues/131

Merging this PR now. 🎉

**Comment by Neme12** [comment]
@jcansdale I can submit another PR to use the new project system, but it has a few disadvantages:

1. There is no designer for `vsixmanifest`
2. <del>In Add -> New Item, there are no items under Extensibility</del> I have a workaround for this
3. VS 15.8 introduces a new feature for extension development which lets you select which Visual Studio instance to deploy to:
image
This wouldn't work. This can of course be worked around (just like today) by specifying this in the .csproj or launchSettings.json, but this would no longer light up automatically based on the user's installations.

It's possible there would be other limitations I'm not aware of. Do you think these limitations are acceptable or would you prefer to stay on the existing project system for the time being?

**Comment by Neme12** [comment]
The approach of using the new project system with custom workarounds seems to be used widely across Microsoft repos but is not supported and it's common to encounter missing features and I would assume also missing integration with 3rd party tools. Basically, "you're on your own".

**Comment by jcansdale** [comment]
Neme12 Thanks for investigating this. Was this one of the projects you were looking at?
https://github.com/tunnelvisionlabs/JustMyCodeToggle/blob/master/Tvl.VisualStudio.JustMyCodeToggle/Tvl.VisualStudio.JustMyCodeToggle.csproj

**Pull request merged**